dhall-kubernetes-generator/src/Dhall/Kubernetes/Convert.hs

```diff
@@ -292,3 +297,56 @@ getImportsMap prefixMap duplicateNameHandler objectNames folder toInclude
         namespaced = case filter filterFn namespacedNames of
           [name] -> Just name
           names  -> duplicateNameHandler (kind, names)
+
+toDefinition :: V1beta1CustomResourceDefinition -> Maybe (ModelName, Definition)
+toDefinition crd = 
+    fmap (\d -> (modelName, d)) definition
+  where
+    spec = v1beta1CustomResourceDefinitionSpec crd
+    group = v1beta1CustomResourceDefinitionSpecGroup spec
+    crdKind = (v1beta1CustomResourceDefinitionNamesKind . v1beta1CustomResourceDefinitionSpecNames) spec
+    modelName = ModelName (group <> "." <> crdKind)
+    definition = do
+      let 
+        versionName xs = case xs of 
+          (x:_) -> Just (v1beta1CustomResourceDefinitionVersionName x)
+          _ -> Nothing
+      version <- 
+        v1beta1CustomResourceDefinitionSpecVersion spec 
+        <|> do 
+          versions <- v1beta1CustomResourceDefinitionSpecVersions spec 
+          versionName versions             
+      validation <- v1beta1CustomResourceDefinitionSpecValidation spec
+      schema <- v1beta1CustomResourceValidationOpenApiv3Schema validation
+      let baseData = BaseData {
+        kind = crdKind,
+        apiVersion = version
+      }
+      let completeSchemaProperties = 
+            fmap 
+            (Data.Map.union
+              (
+                Data.Map.fromList [
+                  ("apiVersion", (mkV1beta1JSONSchemaProps {v1beta1JSONSchemaPropsType = Just "string"}))
```

Comment:
After merging in `master` I believe several of these Kubernetes resource types are now `v1` instead of `v1beta1`, including `JSONSchemaProps` and `CustomResource*`

Reply:
@Gabriel439 Slightly off topic question: 
What version of the API is `dhall-kubernetes` now on? The `swagger.json` in master is still the `v1.14` one i believe. I see that #97 updated the API though. If it's a version above 1.14 i think we will have to fork as AWS is still stuck on 1.14 :( Otherwise #75 will need a solution. I'm asking because CustomResourceDefinition is still v1beta1 in 1.14. 

I don't think this will make any difference anyway as this is reading a yaml CRD file to generate dhall types for the types described by the CRD itself, they are not necessarily coupled. If we need to move to v1 anyway, i'll have to seek to upgrade [kubernetes-client-core](https://hackage.haskell.org/package/kubernetes-client-core) or use hand rolled types (i'm fine with that). Let me know what you think it's the best solution :)

Reply:
@amarrella: We're on version 1.16.3 of `kubernetes`.  The `swagger2.json` in the repo is mis-leading since the build doesn't use it at all.

The build gets the OpenAPI specification from whatever `kubernetes` source repository that `nixpkgs` build against.  Since #97 upgraded `nixpkgs` it also upgraded the version of `kubernetes` we were using to 1.16.3:

https://github.com/NixOS/nixpkgs/blob/3140fa89c51233397f496f49014f6b23216667c2/pkgs/applications/networking/cluster/kubernetes/default.nix#L18

I plan to make a follow-up change to delete the `swagger2.json` file to avoid confusion

Reply:
@Gabriel439: thanks for the clarification. In light of this I think #75 has even more relevance, i'll post my thoughts there to not pollute the PR